### PR TITLE
Add ability to pass pod Annotations and Labels during Worker Pool Launch

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Closes https://github.com/OpenMined/Heartbeat/issues/1334

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- [x] Tests added in notebook

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/grid/helm/syft/values.yaml [hunk 1]
```diff
@@ -159,15 +159,18 @@ node:
...
   defaultBucketName: null
   queuePort: 5556
   logLevel: info
   debuggerEnabled: false
   associationRequestAutoApproval: false
   useInternalRegistry: true
 
+  # Worker pool settings
+  inMemoryWorkers: false
+  defaultWorkerPoolCount: 1
```
Comment (yashgorana) [inline packages/grid/helm/syft/values.yaml]: I think we should split this into

```yaml
defaultWorkerPool:
    tag: null
    count: 1
    podAnnotations: null
    podLabels: null
```

thoughts?

File: packages/syft/src/syft/service/worker/worker_pool_service.py [hunk 2]
```diff
@@ -69,6 +69,7 @@ def launch(
         num_workers: int,
         registry_username: str | None = None,
         registry_password: str | None = None,
+        pod_annotations: dict[str, str] | None = None,
     ) -> list[ContainerSpawnStatus] | SyftError:
```
Comment (yashgorana) [inline packages/syft/src/syft/service/worker/worker_pool_service.py:74]: can we add pod labels as well? label selectors may also be used to place certain pods in certain node pools

File: packages/grid/helm/syft/values.yaml (hunk 1 above)
Comment (rasswanth-s) [reply packages/grid/helm/syft/values.yaml]: Addressed in [9692b6e](https://github.com/OpenMined/PySyft/pull/8884/commits/9692b6e8cbad38bf272f5e66761b6a95b1337e94)

File: packages/syft/src/syft/service/worker/worker_pool_service.py (hunk 2 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/service/worker/worker_pool_service.py:74]: Addressed in [9692b6e](https://github.com/OpenMined/PySyft/pull/8884/commits/9692b6e8cbad38bf272f5e66761b6a95b1337e94)

File: packages/grid/helm/syft/values.yaml [hunk 3]
```diff
@@ -159,15 +159,20 @@ node:
...
   defaultBucketName: null
+  inMemoryWorkers: false
   queuePort: 5556
   logLevel: info
   debuggerEnabled: false
   associationRequestAutoApproval: false
   useInternalRegistry: true
 
+  # Deault Worker pool settings
+  defaultWorkerPool:
```
Comment (yashgorana) [inline packages/grid/helm/syft/values.yaml:171]: @rasswanth-s  we also need to update the following yaml files:
- devspace.yaml
- helm/examples/**/*.yaml
- values.dev.*.yaml

File: packages/syft/src/syft/service/worker/worker_pool_service.py
Comment (yashgorana) [inline packages/syft/src/syft/service/worker/worker_pool_service.py:1]: @IrinaMBejan we've updated the API to include the args `pod_annotations` (dict) & `pod_labels` (dict) to
- `worker_pool.launch`
- `worker_pool.create_pool_request`
- `worker_pool.create_image_and_pool_request`

File: packages/grid/helm/syft/values.yaml (hunk 3 above)
Comment (rasswanth-s) [reply packages/grid/helm/syft/values.yaml:171]: Addressed in [a33d981](https://github.com/OpenMined/PySyft/pull/8884/commits/a33d981f486457ec26601d194d90cc5a638f9fa8)

**Review by yashgorana — APPROVED** [review]
Awaiting @IrinaMBejan's review before merging

**Review by IrinaMBejan — APPROVED** [review]

**Pull request merged**